### PR TITLE
Fix: enable Observability in runtime clusters

### PR DESCRIPTION
**Pull request by zzxwill**
Enabled add-on observability in runtime clusters. In each cluster, a sub-domain
will be allocated to the cluste

Signed-off-by: Zheng Xi Zhou <zzxwill@gmail.com>

### Description of your changes

Fixes #

I have:

- [ ] Read and followed KubeVela's [contribution process](https://github.com/oam-dev/kubevela/blob/master/contribute/create-pull-request.md).
- [ ] [Related Docs](https://github.com/oam-dev/kubevela.io) updated properly. In a new feature or configuration option, an update to the documentation is necessary. 
- [ ] Run `make reviewable` to ensure this PR is ready for review.
- [ ] Added `backport release-x.y` labels to auto-backport this PR if necessary.

### CODE REVIEW

File: pkg/addon/addon.go [hunk 1]
```diff
@@ -25,8 +26,14 @@ import (
 	"strconv"
 	"strings"
 	"sync"
+	"text/template"
```
[the remark this answers is not in the document]
Comment (zzxwill) [reply pkg/addon/addon.go:29]: ignore

File: test/e2e-addon-test/addon_test.go [hunk 2]
```diff
@@ -121,4 +121,15 @@ var _ = Describe("Addon tests", func() {
 			},
 			time.Second*30, time.Millisecond*500).ShouldNot(BeNil())
 	})
+
+	PIt("Addon observability is successfully enabled", func() {
```
Comment (FogDong) [inline test/e2e-addon-test/addon_test.go:125]: Why PIT this?

**Review by FogDong — APPROVED** [review]

**Review by Somefive — APPROVED** [review]
I would recommend #ApplyApplicationInParallel to add a post check after parallel dispatch.

File: pkg/addon/addon.go [hunk 3]
```diff
@@ -489,7 +543,32 @@ func RenderApp(addon *Addon, config *rest.Config, args map[string]interface{}) (
 				Name: "deploy-runtime",
 				Type: "deploy2runtime",
 			})
-	} else {
+	case addon.Name == "observability":
+		arg, ok := args["domain"]
```
Comment (wangyikewxgm) [inline pkg/addon/addon.go:547]: Please create an issue and assign me，we should  use some  mechanism  to refactor these hack code.

File: pkg/addon/addon.go [hunk 4]
```diff
@@ -758,15 +930,15 @@ func (h *Handler) checkDependencies() error {
...
+		if err = depHandler.enableAddon(ctx, k8sClient); err != nil {
 			return errors.Wrap(err, "fail to dispatch dependent addon resource")
 		}
 	}
 	return nil
 }
 
-func (h *Handler) dispatchAddonResource() error {
-	app, err := RenderApp(h.addon, h.config, h.args)
+func (h *Handler) dispatchAddonResource(ctx context.Context, k8sClient client.Client) error {
```
Comment (wangyikewxgm) [inline pkg/addon/addon.go:940]: why we should refactor this func, we can get context and cli from handler.

File: test/e2e-addon-test/addon_test.go (hunk 2 above)
Comment (zzxwill) [reply test/e2e-addon-test/addon_test.go:125]: Addon observability is not allowed to be integrated to `oam-dev/catalog/addons` until this PR passed. So I have to escape this test and enable it later.

File: pkg/addon/addon.go (hunk 3 above)
Comment (zzxwill) [reply pkg/addon/addon.go:547]: https://github.com/oam-dev/catalog/issues/193

File: pkg/addon/addon.go (hunk 4 above)
Comment (zzxwill) [reply pkg/addon/addon.go]: Fixed, thanks.

**Review by hongchaodeng — APPROVED** [review]

**Pull request merged**